### PR TITLE
feat(nuxt): add hook debug mode

### LINKED ISSUE #3718: debug hooks
Nuxt has a `debug` option which is currently unused by Nuxt 3. We could use this flag to show verbose debug logs and also debug hooks and their timing  (upstream: unjs/hookable#37)

_(inspired from mention by @danielroe)_

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

resolves #3718

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR adds support for `debug: true` in your nuxt config (disabled by default). At the moment, it prints out hook names and timings on the server, and logs hook arguments as well in the browser.

#### Example output

**Server**

```bash
[nuxt] app:created: 2.109ms
[nuxt] vue:setup: 0.035ms
[nuxt] app:rendered: 0.019ms
[nitro] render:html: 0.048ms
[nitro] render:response: 0.01ms
```

**Client**

CleanShot 2022-09-20 at 16 33 23

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by pi0** [comment]
This is a nice improvement but what do you think if we directly add `hookable.setupDebugger(hooks, { tag })`? We have at least 4 hookable namespace now (nuxt build, nuxt app, nitro build and nitro app). It can consistently add feature.

**Comment by danielroe** [comment]
I like the idea. What do you think about enabling a custom handler as well (or 'presets') as I think it's nicer to have a browser API with collapsed/grouped logs and also allow logging all args, which would be unfeasibly verbose in the terminal?

**Comment by pi0** [comment]
Makes sense to inspect only for browser. Moved to #https://github.com/unjs/hookable/issues/50.

**Comment by danielroe** [comment]
Are we waiting for anything in https://github.com/unjs/hookable/pull/51?

**Comment by pi0** [comment]
hookable@5.4.0 released

File: packages/nuxt/src/core/nitro.ts [hunk 1]
```diff
@@ -144,6 +150,13 @@ export async function initNitro (nuxt: Nuxt & { _nitro?: Nitro }) {
 
   // Connect hooks
   nuxt.hook('close', () => nitro.hooks.callHook('close'))
+  if (nuxt.options.debug) {
+    // @ts-expect-error remove in next version of hookable
+    const { close } = createDebugger(nitro.hooks, {
```
Comment (pi0) [inline packages/nuxt/src/core/nitro.ts:155]: We might directly move it to nitro for benchmarking hooks

Comment (danielroe) [reply packages/nuxt/src/core/nitro.ts:155]: on it

Comment (danielroe) [reply packages/nuxt/src/core/nitro.ts]: ~> https://github.com/unjs/nitro/pull/577

**Review by pi0 — APPROVED** [review]

**Comment by danielroe** [comment]
I had some other changes I was going to make when hookable was released. Will check later.

**Pull request merged**